sagemath: update to 10.3.

tornaria opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

PR for tracking sagemath beta/rc until 10.3 is released.

#### Testing the changes
- I tested the changes in this PR: **briefly**

<!--
#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**|**NO**
-->

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->
<!--
#### Local build testing
- I built this PR locally for my native architecture, (ARCH-LIBC)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv6l-musl
-->

dkwo commented on srcpkgs/sagemath/template:
```diff
@@ -26,14 +26,14 @@ depends="eclib-devel fflas-ffpack flintlib-devel gcc-fortran gd-devel
 distfiles="https://github.com/sagemath/sage/archive/refs/tags/$version.tar.gz"
-checksum=e7125f13495e1068edab73735aca7f9b2c655688096e9d109e628c023e76411f
+checksum=3f72b8d13d8bb472daaeabff4feba98e392028328e20012980da9c457dc49fb0
 nocross="due to ntl (eclib), fflas-ffpack, givaro, linbox, sympow, maxima"
```
Can you add `singular` here, just so we don't forget?
`nocross="due to ntl (eclib, singular), fflas-ffpack, givaro, linbox, sympow, maxima"`

tornaria replied:
But singular cross builds just fine, doesn't it? Is the cross-built singular not suitable to build sagemath?

dkwo replied:
indeed, the cross-built singular (i.e. built without ntl) gives a lot of segfaults in sage doctests.

dkwo replied:
yes, i agree.
sounds good.

tornaria commented:
@ahesford testing here cython 3.0.7 to compile and check sagemath

ahesford commented:
Thanks for checking Cython; updated in 0dea09d39debbab95eeb01110423a619e370498c.

tornaria commented:
@dkwo sagemath 10.3 is close to being released. This PR is up-to-date with rc1, using flintlib 3.1 (it's just simpler to update both at the same time -- singular needs patching as well).

Also I changed (slightly) the package, first to use pep517 build (which works fine) and also moved the internal scripts from `/usr/lib/sagemath/10.3/bin` directly into `/usr/libexec/sagemath`. Both changes should be transparent to the user.

If you want to test (esp. in aarch64) let me know of any trouble. We have a small window to contribute fixes upstream if something pops up.

tornaria commented:
I added the patch to givaro; revbump to be safe since the missing include is in a header that we ship in givaro-devel.

tornaria commented:
I'm getting weird "error" in CI (no log at all), I don't know what is going on...

tornaria commented:
It turns out that the `flintlib-3.1.0.tar.gz` tarball had changed (to fix two issues I reported, see https://github.com/flintlib/flint/issues/1809) hence the template was failing at fetching it. It's updated now.

tornaria commented:
> I'm getting weird "error" in CI (no log at all), I don't know what is going on...
It seems the logs are now XHRed from a `*.windows.net` address instead of `*.github.com`, which umatrix was blocking. This confused me as it seemed the CI itself was completely broken.

dkwo commented:
It all builds and passes checks on aarch64 now, thanks!

tornaria commented:
@dkwo final release of sagemath 10.3, if you want to review and give one more try before we merge.

Notes
- gap: I'm holding back 4.13.0 since there is no PR available on sagemath yet (https://github.com/sagemath/sage/issues/37616)
- singular: updated to 4.3.2p16 for flint 3.1 support, needs a small https://github.com/sagemath/sage/pull/37492; I added myself as co-maintainer
- switched to pep517 build, but still from a github tarball; to build from pypi sdist we are missing https://github.com/sagemath/sage/pull/37287

dkwo commented:
This builds and checks fine on aarch64 :)
time to merge it? @leahneukirchen

leahneukirchen commented:
Then mark as ready.

tornaria commented:
Ready to merge! Thanks @dkwo @leahneukirchen for your support.
